src/app/Library/CrudPanel/Traits/SaveActions.php

```diff
@@ -15,52 +19,260 @@ public function getSaveActionDefaultForCurrentOperation()
+     *
+     * @param int $wantedOrder
+     * @return int
+     */
+    public function reorderSaveActions($wantedOrder)
+    {
+        if (! empty($this->availableSaveActions)) {
+            $lastOrder = max(array_column($this->availableSaveActions, 'order'));
+            foreach ($this->availableSaveActions as &$sv) {
+                if ($wantedOrder == $sv['order']) {
+                    if (empty(array_filter($this->availableSaveActions, function ($arr) {
+                        $arr['order'] == 1;
+                    }))) {
+                        $sv['order'] = 1;
+                    } else {
+                        $sv['order'] = $lastOrder + 1;
+                    }
+                }
+            }
+        }
+
+        return $wantedOrder;
+    }
+
+    /**
+     * Replace the current save actions with the ones provided.
+     *
+     * @param array $saveActions
+     * @return void
+     */
+    public function replaceSaveActions($saveActions)
+    {
+        if (is_array($saveActions)) {
+            //we reset all save actions
+            $this->availableSaveActions = [];
+
+            $this->applyRegisterSaveActions($saveActions);
+        }
+    }
+
+    /**
+     * Allow the developer to unset save actions.
+     *
+     * @param string $saveAction
+     * @return void
+     */
+    public function forgetSaveActions($saveAction)
+    {
+        if (is_array($saveAction)) {
+            foreach ($saveAction as $sv) {
+                if (isset($this->availableSaveActions[$sv])) {
+                    unset($this->availableSaveActions[$sv]);
+                }
+            }
+        } else {
+            if (isset($this->availableSaveActions[$saveAction])) {
+                unset($this->availableSaveActions[$saveAction]);
+            }
+        }
+    }
+
+    /**
+     * Allow the developer to unset all save actions.
+     *
+     * @param string $saveAction
+     * @return void
+     */
+    public function forgetAllSaveActions()
+    {
+        $this->availableSaveActions = [];
+    }
+
+    /**
+     * Checks if a save action exists.
+     *
+     * @param string $saveAction
+     * @return bool
+     */
+    public function saveActionExists($saveAction)
+    {
+        return isset($this->availableSaveActions[$saveAction]);
+    }
+
+    /**
+     * Apply the orders to save actions array.
+     *
+     * @param string $saveAction
+     * @param int $order
+     * @return void
+     */
+    public function applyOrderToSaveAction($saveAction, $order)
+    {
+        if ($this->saveActionExists($saveAction)) {
+            $this->reorderSaveActions($order);
+
+            $this->availableSaveActions[$saveAction]['order'] = $order;
+        }
+    }
+
+    /**
+     * Allows the developer to set save actions order.
+     *
+     * @param string|array $saveAction
+     * @param int|null $order
+     * @return void
+     */
+    public function orderSaveActions($saveAction, $order = null)
```

Comment:
@pxpm could you please explain the syntax you had in mind for this? How will the developer call this?

I see for Fields we have ```orderFields($order)``` which can be called to reorder them by giving an array of all fields in the right order (so basically saying "this is the right order from now on"). I'm not saying that's the way to go, I'm just pointing out how we're doing the same thing, somewhere else.

Reply:
Forget you saw this code. :smiling_imp: 
I do this PR's mainly as sketches to see if it works, if it's how you'd expect things to go and simpler/fast solutions I find for the problems. I'd happily refactor later :+1: 

Sooo i refactored the function into two singular/plural.

The function should be used: `orderSaveAction('save_action_name',integer_desired_order')`, or `orderSaveActions(['save_action_1' => 1, 'save_action_2' => 2])`

To keep coherency I added the ability to also do like in Fields `orderSaveActions(['save_action_1', 'save_action_2'])`